python/tests/test_tables.py

```diff
@@ -2655,3 +2655,118 @@ def test_empty_nodes(self):
             self.assertEqual(subset.sites.num_rows, 0)
             self.assertEqual(subset.mutations.num_rows, 0)
             self.assertEqual(subset.provenances, tables.provenances)
+
+
+class TestUnion(unittest.TestCase):
+    """
+    Tests for the TableCollection.union method
+    """
+
+    def get_msprime_example(self, sample_size=3, T=5, seed=1239):
+        # we assume after the split the ts are completely independent
+        M = [[0, 0], [0, 0]]
+        population_configurations = [
+            msprime.PopulationConfiguration(sample_size=sample_size),
+            msprime.PopulationConfiguration(sample_size=sample_size),
+        ]
+        demographic_events = [
+            msprime.CensusEvent(time=T),
+            msprime.MassMigration(T, source=1, dest=0, proportion=1),
+        ]
+        ts = msprime.simulate(
+            population_configurations=population_configurations,
+            demographic_events=demographic_events,
+            migration_matrix=M,
+            length=2e5,
+            recombination_rate=1e-8,
+            mutation_rate=1e-7,
+            record_migrations=False,
+            random_seed=seed,
+        )
+        ts = tsutil.add_random_metadata(ts, seed)
+        ts = tsutil.insert_random_ploidy_individuals(ts, max_ploidy=1)
+        return ts
+
+    def get_wf_example(self, N=5, T=5, seed=1249):
+        twopop_tables = wf.wf_sim(N, T, num_pops=2, seed=seed, deep_history=True)
+        twopop_tables.sort()
+        ts = twopop_tables.tree_sequence()
+        ts = ts.simplify()
+        # adding muts
+        ts = tsutil.jukes_cantor(ts, 1, 10, seed=seed)
+        ts = tsutil.add_random_metadata(ts, seed)
+        ts = tsutil.insert_random_ploidy_individuals(ts, max_ploidy=2)
+        return ts
+
+    def split_example(self, ts, T):
+        # splitting two pop ts into disjoint ts
+        shared_nodes = [n.id for n in ts.nodes() if n.time >= T]
+        pop1 = list(ts.samples(population=0))
+        pop2 = list(ts.samples(population=1))
+        tables1 = ts.simplify(shared_nodes + pop1, record_provenance=False).tables
+        tables2 = ts.simplify(shared_nodes + pop2, record_provenance=False).tables
+        node_mapping = [
+            i if i < len(shared_nodes) else tskit.NULL
+            for i in range(tables2.nodes.num_rows)
+        ]
+        return tables1, tables2, node_mapping
+
+    def get_examples(self, N=5, T=5, seed=242):
+        yield self.split_example(self.get_msprime_example(N, T, seed=seed), T)
```

Comment:
I don't think this indirection helps much. It would be easier to just call these two functions directly below where they are used (and also could be registered directly then as subtests).

Reply:
agree